linkerd/app/outbound/src/lib.rs

```diff
@@ -485,37 +487,32 @@ impl Config {
                     .box_http_response(),
             )
             .push(svc::layer::mk(http::normalize_uri::MakeNormalizeUri::new))
-            .instrument(|_: &listen::Addrs| debug_span!("source"))
-            .check_new_service::<listen::Addrs, http::Request<_>>()
-            .into_make_service()
+            .instrument(|_: &_| debug_span!("source"))
+            .push_map_target(endpoint::HttpAccept::from)
+            .check_new_service::<(http::Version, endpoint::TcpLogical), http::Request<_>>()
             .into_inner();
 
         let tcp_forward = svc::stack(tcp_connect.clone())
             .push_make_thunk()
             .push_on_response(svc::layer::mk(tcp::Forward::new))
             .instrument(|_: &TcpEndpoint| info_span!("forward"))
-            .check_service::<TcpEndpoint>();
+            .check_new::<TcpEndpoint>();
 
         // Load balances TCP streams that cannot be decoded as HTTP.
         let tcp_balance = svc::stack(self.build_tcp_balance(tcp_connect, resolve))
             .push_fallback_with_predicate(
-                svc::stack(tcp_forward.clone())
-                    .check_new::<TcpEndpoint>()
+                tcp_forward
+                    .clone()
                     .push_map_target(TcpEndpoint::from)
                     .into_inner(),
                 is_discovery_rejected,
             )
-            .cache(
-                svc::layers().push_on_response(
-                    svc::layers()
-                        .push_failfast(dispatch_timeout)
-                        .push_spawn_buffer_with_idle_timeout(buffer_capacity, cache_max_idle_age),
-                ),
+            .push_on_response(
+                svc::layers()
+                    .push_failfast(dispatch_timeout)
+                    .push_spawn_buffer_with_idle_timeout(buffer_capacity, cache_max_idle_age),
             )
-            .into_make_service()
-            .spawn_buffer(buffer_capacity)
-            .instrument(|_: &_| info_span!("tcp"))
```

Comment:
nit: do we still want the instrument layer here?

Reply:
the tcp scope is now set in detect-http. it's probably better to stop doing that and keep them in here